### PR TITLE
Added suppport for combining multiple feeds in one request

### PR DESCRIPTION
**Pull request by joaquinloustau**
@rachelriv if you look closely, `twitter-router` in stream-api directory is different from the one in `social-media-api` directory. We are making use of a different api call, and using different oauth credentials, so that's why there's a different directory -separation of concerns. If we decide to go this route for our twitter api calls, the `social-media-api` can be deprecated in its entirety.

### CODE REVIEW

**Comment by rachelriv** [comment]
It looks like you copied the `facebook-router`, `twitter-router` and `server` into another directory. All three of these files still exist in their original directory, so is there a reason why you didn't use the original files?

File: processing/streamtweetlogging.coffee [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    tweets = JSON.parse body
+    if err || tweets.errors?
+      console.log 'there was an error in the request'
+      console.log err 
+      console.log tweets.errors
+
+    for tweet in tweets
+      parsedTweet = chrono.parse tweet.text
+      if parsedTweet.length != 0
```
Comment (rachelriv) [inline processing/streamtweetlogging.coffee:25]: I think `if parsedTweet.length isnt 0`is more idiomatic CoffeeScript

**Comment by rachelriv** [comment]
Do we have any tests for this?

File: processing/streamtweetlogging.coffee [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+fbURL = ''
+
+twitterProcessing = (listID, sinceID) ->
+  requestURL = "#{serverName}#{twitterURL}#{listID}"
+  requestURL += sinceID if sinceID
+
+  request requestURL , (err, res, body) ->
+
+    tweets = JSON.parse body
+    if err || tweets.errors?
```
Comment (rachelriv) [inline processing/streamtweetlogging.coffee:18]: `or` instead of `||`

File: processing/twitter-events.coffee [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+twitterProcessing = (screenName, sinceID) ->
+  console.log 'screenName: ' + screenName
+  requestURL = "#{serverName}#{twitterURL}#{screenName}"
+  console.log 'requestURL: ' + requestURL
+
+  request requestURL , (err, res, body) ->
+    console.log 'err: ' + err
+    console.log 'body: ' + body
+    tweets = JSON.parse body
+    console.log 'tweets: ' + tweets
```
Comment (rachelriv) [inline processing/twitter-events.coffee:26]: Is it necessary to leave all of these `console.log`s in here?

File: processing/twitter-events.coffee [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    if err || tweets.errors?
```
Comment (rachelriv) [inline processing/twitter-events.coffee:27]: `or` instead of `||`

File: processing/twitter-events.coffee [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      console.log 'there was an error in the request'
+
+    for tweet in tweets
+      parsedTweet = chrono.parse tweet.text
+    
+      if parsedTweet.length != 0
```
Comment (rachelriv) [inline processing/twitter-events.coffee:33]: `isnt` can be used

**Comment by rachelriv** [comment]
Do you think we should keep the `log.txt` in here? It is the file generated by the processor, and should be saved on a local machine, but I don't think it is appropriate to have it in our repository.

File: processing/twitter-events.coffee (hunk 3 above)
Comment (joaquinloustau) [inline processing/twitter-events.coffee:26]: There is another commit in the merge pull request that got rid of them

**Comment by joaquinloustau** [comment]
I just pushed the `log.txt` and `log2.txt` so it can be accessed by @juansc  and @cbillingham  for training our nlp purposes. Note that this merge is not into master but into a development branch, (jsc/processing-posts). Therefore, when someone runs the script let's say tonight and gets new events, they can push the file with the new events. And if I were to run the script tomorrow I would just pull the latest version of the file and see when was the last time that  Juan or Cameron fetched events.

File: processing/streamtweetlogging.coffee (hunk 1 above)
Comment (joaquinloustau) [inline processing/streamtweetlogging.coffee:25]: Noted

File: processing/streamtweetlogging.coffee (hunk 2 above)
Comment (joaquinloustau) [inline processing/streamtweetlogging.coffee:18]: Noted

File: processing/twitter-events.coffee (hunk 4 above)
Comment (joaquinloustau) [inline processing/twitter-events.coffee:27]: Noted

File: processing/twitter-events.coffee (hunk 5 above)
Comment (joaquinloustau) [inline processing/twitter-events.coffee:33]: Noted

**Comment by juansc** [comment]
I think it's fine to have in this branch. It will be for testing purposes, and when we merge into master we can destroy the log files.

**Comment by rachelriv** [comment]
I think it is more conventional to alter the existing router files than to copy the files and make changes on the copies. That is what we have version control for :smile:

**Comment by rachelriv** [comment]
@juansc @joaquinloustau Automatically generated files such as log files or files produced by the build system are typically kept in a `.gitignore`. Even if we "destroy" the log files when merging into master, the files will still exist in the repository's history. I think it might be good to ignore these kinds of files moving forward.

**Comment by juansc** [comment]
I don't see this as a log file full of meaningless data that changes each time we do a build. This is a compilation of tweets that we can use later to train the classifier and also so we can test the date processing without having to make needless API calls. So I don't think it's wrong to keep it here.

**Comment by joaquinloustau** [comment]
@rachelriv @juansc  I agree that they are typically kept in a `.gitignore` file, but I think that while we set up our database, these files are the only way of keeping a record of events. And unless we just have one member of the team be the only one fetching events daily or even multiple times a day (we have a limit of 100 tweets for over 50 feeds) I think it is necessary for everyone in the team to have access to it and have it in the repository. Once we have the database setup, there will be no need for these files.  We can further discuss it tomorrow

**Comment by joaquinloustau** [comment]
I agree with Juan. If the naming convention for the file has led to confusion, I am more than happy to change its name.

**Comment by rachelriv** [comment]
Okay yeah I was a bit confused because of the name. Since the `log.txt` isn't what is typically thought of as a log file, then it would be good to change the name to maybe `aggregated-test-posts.txt` or something. However, I do think that we should be making changes to existing code instead of copying code into a separate file and making changes to the copied file.

**Comment by joaquinloustau** [comment]
I think both files need to coexist for the time being until we make a decision, but we can further discuss it tomorrow @rachelriv.

**Comment by rachelriv** [comment]
@joaquinloustau I think that the whole idea of having a version control system like this is that it allows you to make changes as you see fit. The beauty of git is that we can always choose to not merge in your changes, or even if we do, we can revert back to an earlier version. Making a copy of the file and altering the copy is not how [GitHub flow](https://guides.github.com/introduction/flow/) works. Additionally, it introduces a lot of unnecessary repetition into our repository. "Clean Code" by Uncle Bob, our reading assignment from @rtoal, talks about the importance of the DRY principle (Don't Repeat Yourself). 

As far as using different oauth credentials, you should be able to just change the credentials within the `.env` file of the `social-media-api` directory. :smiley:

**Comment by cbillingham** [comment]
sorry. just floating on github and I saw this. Can we close this request?

**Pull request closed without merging**